xinference/core/chat_interface.py

```diff
@@ -12,42 +12,56 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import logging
 import os
-from typing import Dict, List
+from typing import Generator, List
 
 import gradio as gr
 from gradio.components import Markdown, Textbox
 from gradio.layouts import Accordion, Column, Row
 
-from ..client import RESTfulClient
+from ..client.restful.restful_client import (
+    RESTfulChatModelHandle,
+    RESTfulGenerateModelHandle,
+)
+from ..types import ChatCompletionMessage
+
+logger = logging.getLogger(__name__)
 
 
 class LLMInterface:
     def __init__(
         self,
         endpoint: str,
         model_uid: str,
+        model_name: str,
+        model_size_in_billions: int,
+        model_format: str,
+        quantization: str,
+        context_length: int,
+        model_ability: List[str],
+        model_description: str,
+        model_lang: List[str],
     ):
-        self.client = RESTfulClient(endpoint)
         self.endpoint = endpoint
         self.model_uid = model_uid
+        self.model_name = model_name
+        self.model_size_in_billions = model_size_in_billions
+        self.model_format = model_format
+        self.quantization = quantization
+        self.context_length = context_length
+        self.model_ability = model_ability
+        self.model_description = model_description
+        self.model_lang = model_lang
 
     def build(self) -> "gr.Blocks":
-        model = self.client.get_model(self.model_uid)
-        model_info = self.client.describe_model(self.model_uid)
-        model_ability = model_info["model_ability"]
-
-        if "chat" in model_ability:
-            interface = self.build_chat_interface(
-                model,
-                model_info,
-            )
+        if "chat" in self.model_ability:
+            interface = self.build_chat_interface()
         else:
-            interface = self.build_generate_interface(
-                model,
-                model_info,
-            )
+            interface = self.build_generate_interface()
 
+        interface.queue()
+        interface.startup_events()
```

Comment:
Plz add a comment here to indicate why should use `startup_events` method for better reading. Link to that issue.